src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -248,15 +236,7 @@ abstract class BTypes {
      *  - for an ARRAY type, the full descriptor is part of the range
      */
     def toASMType: asm.Type = this match {
-      case UNIT   => asm.Type.VOID_TYPE
-      case BOOL   => asm.Type.BOOLEAN_TYPE
-      case CHAR   => asm.Type.CHAR_TYPE
-      case BYTE   => asm.Type.BYTE_TYPE
-      case SHORT  => asm.Type.SHORT_TYPE
-      case INT    => asm.Type.INT_TYPE
-      case FLOAT  => asm.Type.FLOAT_TYPE
-      case LONG   => asm.Type.LONG_TYPE
-      case DOUBLE => asm.Type.DOUBLE_TYPE
+      case p: PrimitiveBType        => p.asmType
       case ClassBType(internalName) => asm.Type.getObjectType(internalName) // see (*) above
```

Comment:
ClassBtype is interned, so we should cache this in the ClassBType
https://github.com/scala/scala/pull/8456/files#r331297400
e.g. some types will be referred to lots